tests/integration_tests/build/test_dependencies.py

```diff
@@ -24,4 +24,17 @@ def test_licenses():
         os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../../Cargo.toml")
     )
 
-    cargo("deny", f"--manifest-path {toml_file} check licenses bans")
+    _, stdout, stderr = cargo("deny", f"--manifest-path {toml_file} check licenses bans")
+    assert "licenses ok" in stdout
+
+    # "cargo deny" should deny licenses by default but for some reason copyleft is allowed
+    # by it and if we add a dependency which has copyleft licenses "cargo deny" won't report
+    # it unless it is explicitly told to do so from the deny.toml.
+    # Our current deny.toml seems to cover all the cases we need but,
+    # if there is an exception like copyleft (where we don't want and don't deny
+    # in deny.toml and is allowed by cardo deny), we don't want to be left in the dark.
+    # For such cases check "cargo deny" output, make sure that there are no warnings reported
+    # related to the license and take appropriate actions i.e. either add them to allow list
+    # or remove them if they are incompatible with our licenses.
+    license_res = [line for line in stderr.split("\n") if "license" in line]
```

Comment:
was there a problem with just asserting stderr is empty? I still think that would be a more robust solution, say if the exact output of the tool ever changes.

Reply:
we have a lot of other warning related to duplicate entries in Cargo.lock e.g. 
```
E   warning[duplicate]: found 2 duplicate entries for crate 'windows_x86_64_msvc'
E       ┌─ /firecracker/Cargo.lock:194:1
E       │  
E   194 │ ╭ windows_x86_64_msvc 0.48.5 registry+https://github.com/rust-lang/crates.io-index
E   195 │ │ windows_x86_64_msvc 0.52.0 registry+https://github.com/rust-lang/crates.io-index
E       │ ╰────────────────────────────────────────────────────────────────────────────────^ lock entries
```

Reply:
Ah, thanks for the explanation! Didnt see those because I only ran `deny check license`, not `license bans`